saltboot-formula/_states/saltboot.py

```diff
@@ -1911,13 +1925,15 @@ def verify_and_boot_system(name, partitioning, images, boot_images, action = 'fa
         report_progress = __salt__['file.is_fifo']("/progress")
 
     newroot = __salt__['environ.get']("NEWROOT", default="/mnt")
-
     saltboot_fallback = __salt__['environ.get']('saltboot_fallback')
+
+    salt_pid = _get_salt_pid()
+
```

Comment:
Can you move this call to the `_request_salt_stop` function? Calls below will not need any argument and all will be nicely together.